File: backend/src/main/java/com/fdb/backend/Entities/Announcement.java
package com.fdb.backend.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Entity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Announcement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int announcementId;

    private String content;

    private LocalDate date;

    private LocalTime time;

    @ManyToOne
    @JoinColumn(name = "course_id")
    private Course course;

    @ManyToOne
    private User user;

    // Constructors, getters, and setters
}

File: frontend/src/app/models/announcementModel/announcement.ts
import { Course } from "../courseModel/course";
import { User } from "../userModel/user";

export class Announcement {
  announcementId!: number;
  content!: string;
  date!: string; 
  time!: string;
  course!: Course; 
  user!: User; 
}
